hyperspy/_signals/spectrum.py

```diff
@@ -84,7 +84,10 @@ def _spikes_diagnosis(self, signal_mask=None,
             dc = dc[~navigation_mask, :]
         der = np.abs(np.diff(dc, 1, -1))
         plt.figure()
-        plt.hist(np.ravel(der.max(-1)), 100)
+        if len(dc.shape) == 1:
```

Comment:
I think by just changing the `ylim` to -1 it should be good enough. Remember this is in log scale still, so leaving it as 0 just cut off any bins with just one count (you can _barely_ see them in the figures you attached... in some cases at least).